### ISSUE TITLE
Tf Lite Micro Global Average Pooling not supported

### ISSUE THREAD

**Issue by DominikStachura** [tensorflow/tensorflow#43332]
@tensorflow/micro

**System information**
- Project build on windows, cygwin console:
- TensorFlow installed from source
- Tensorflow version: 2.3.0, commit 4230dd89cff
- Target platform: stm32f429, ARM4

I use quantized TF Lite model trying to run it on target platform. Model loads correctly, tenaor allocates correctly, but I have a problem during invoking. In my original Keras model I use Global Average Pooling, which is being converted to Mean operation in Tflite file. Netron diagram:

![image](https://user-images.githubusercontent.com/58625554/93563695-b40f6e00-f988-11ea-9a10-3e754cf3cdf5.png)

After invoking I get folowing error:
_tensorflow/lite/micro/kernels/reduce.cc Number of Input dimensions != 4 OR the Axis is not either [1, 2] or [2, 1]

Node MEAN (number 19) failed to invoke with status 1_

I belive that there is a problem in reduce.cc file and MEAN operation implementation. I am not sure why should I provide 4D input for MEAN operation.
Do you plan to implement 3D impleentation or may suggest any workaround for this problem?

Fault file: 
tensorflow_src\tensorflow\lite\micro\kernels\reduce.cc
In EvaMean function:
`TF_LITE_ENSURE_MSG(
      context, is_valid_inputs == true,
      "Number of Input "
      "dimensions != 4 OR the Axis is not either [1, 2] or [2, 1]");`

**Comment by njeffrie** [comment]
I've created an internal change to fix this. Hopefully it should be submitted by early next week.

**Comment by njeffrie** [comment]
Update: This should now be fixed with my change to tensorflow/lite/micro/kernels/reduce.cc.

**Comment by njeffrie** [comment]
https://github.com/tensorflow/tensorflow/commit/9a662b14eaa85ce466be82abd758a636c1eaa756
